lib/core/engine/iteration.js

```diff
@@ -87,8 +88,18 @@ class Iteration {
       await video.setupDirs(index);
     }
 
+    // Instantiate a TCPDUMP instance
+    const myTCPDUMP = new TCPDUMP(
+      this.storageManager.directory,
+      this.options,
+      index
+    );
+
     log.info('Testing url %s iteration %s', url, index);
     try {
+      // Start TCPDUMP
+      myTCPDUMP.start();
```

Comment:
Can you add the check useTcpDump === true here? I think we most try to do the check first before using the function (could be that we have missed that sometimes), think it makes the code easier to read, you could follow the full flow in Iteration.